### PR TITLE
Beta mgmt policy

### LINKED ISSUE #136: Crossplane 1.15 trying to use Management Policies
### What happened?

I'm trying to import a gitlab group so it can be utilized by references

```
apiVersion: groups.gitlab.crossplane.io/v1alpha1
kind: Group
metadata:
  name: zys
  annotations:
    crossplane.io/external-name: 'zys'
spec:
  managementPolicies: ['Observe']
  forProvider:
    path: 'xyz/zys'

```

But it get this status:

```
atProvider: {}
conditions:
  - lastTransitionTime: 2024-04-25T12:20:39Z
    message: "`spec.managementPolicies` is set to a non-default value but the
      feature is not enabled: [Observe]"
    reason: ReconcileError
    status: "False"
    type: Synced
```

I believe that is caused by the Gitlab still using the `EnableAlphaManagementPolicies` instead of the `EnableBetaManagementPolicies` now that it's gone into Beta.

### How can we reproduce it?

Try and import an existing group with the Observe managementPolicy set.

### What environment did it happen in?
Crossplane version:  1.15.2
Crossplane Provider GitLab version: 0.6.1

$ kubectl version
Client Version: v1.30.0
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.29.3-eks-adc7111

### PR DESCRIPTION
**Pull request by jwhitcraft**
### Description of your changes

This enabled the using the Beta Management Policy Feature Flag

Fixes #136

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Built and tested in our test controlplane

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by MisterMX** [comment]
The provider uses https://github.com/crossplane/crossplane-runtime `v1.14.x` which is using the alpha feature flag. As long as we don't make the upgrade to v1.15 this change wont work.

**Comment by jwhitcraft** [comment]
> The provider uses https://github.com/crossplane/crossplane-runtime `v1.14.x` which is using the alpha feature flag. As long as we don't make the upgrade to v1.15 this change wont work.

From the looks of it, the feature went beta in 1.14.0rc1: https://github.com/crossplane/crossplane-runtime/commit/55a8c109304f3a03509216fb366230d965236c40

File: cmd/provider/main.go [hunk 1]
```diff
@@ -122,8 +122,8 @@ func main() {
 	}
 
 	if *enableManagementPolicies {
-		o.Features.Enable(features.EnableAlphaManagementPolicies)
-		log.Info("Alpha feature enabled", "flag", features.EnableAlphaManagementPolicies)
+		o.Features.Enable(feature.EnableBetaManagementPolicies)
+		log.Info("Beta Management Policies feature enabled", "flag", feature.EnableBetaManagementPolicies)
```
Comment (MisterMX) [inline cmd/provider/main.go:126]: `features.EnableAlphaManagementPolicies` is a provider feature flag that does not come with the runtime. Changing this wont work.

Comment (jwhitcraft) [reply cmd/provider/main.go:126]: oh i see, i guess i'll go back to the drawing board and see if i can get this provided updated to 1.15.x so we can use the beta feature flag.

**Comment by MisterMX** [comment]
Closing this, since the underlying issue #136 is solved.

**Pull request closed without merging**